lib/erlef/blogs/repo.ex

```diff
@@ -1,7 +1,39 @@
 defmodule Erlef.Blogs.Repo do
-  def all(repo) do
-    repo.all()
+  @root "priv/posts/**"
+
+  use Nabo.Repo, root: @root
+
+  @spec for_wg(String.t()) :: [Nabo.Post.t()]
+  def for_wg(wg) do
+    load_all()
+    |> filter_by_category(wg)
   end
 
-  def get(repo, id), do: repo.get(id)
+  @spec load_all(String.t()) :: [Nabo.Post.t()]
+  def load_all(path \\ @root) do
+    path
+    |> Path.join(["*", ".md"])
+    |> Path.wildcard()
+    |> Enum.reduce([], &compile/2)
+  end
+
+  defp for_slug(posts, slug) do
+    posts
+    |> Enum.find(&(&1.slug == slug))
+  end
+
+  defp filter_by_category(posts, category) do
+    posts
+    |> Enum.filter(&(Map.get(&1.metadata, "category") == category))
+  end
+
+  defp compile(path, acc) do
+    path
+    |> File.read!()
+    |> Nabo.Compiler.compile([])
+    |> case do
+      {:ok, _slug, post} -> [post | acc]
```

Comment:
Do you think this case expression should go in a function?

Reply:
I think the logic is simple enough that the locality of having the thing right here is beneficial

Reply:
Could assign to a var then do the case expression, but it's working, there's always time to refactor later.